R/finalize_plot.R

```diff
@@ -434,3 +463,75 @@ finalize_plot <- function(input_plot = NULL,
     return(final_plot)
   }
 }
+
+
+
+#' @noRd
+# Function to create plot object with left aligned legend on top
+buildChart <- function(input_plot,
+                       plot_constants,
+                       legend_build) {
+
+  # in safe mode, don't extract legend
+  if(legend_build == "safe"){
+    return(ggplotGrob(input_plot))
+  }
+
+  # in no legend mode, remove legend altogether
+  if(legend_build == "none"){
+    output_plot <- input_plot + theme(legend.position = "none")
+    return(ggplotGrob(output_plot))
+  }
+
+  # Reformat plot
+  ## IS ANY OF THIS ACTUALLY NECESSARY? CAN WE MOVE MORE OF THIS OVER TO THEME_CMAP?
+  format_plot <- input_plot + theme(
+    # ensure legend is left aligned
+    legend.position = "left",
+    # update margins to account for possible indent
+    plot.margin = grid::unit(plot_constants$padding_plot,"bigpts"),
+    legend.margin = margin(t = plot_constants$padding_legend[1],
+                           r = plot_constants$padding_legend[2],
+                           b = plot_constants$padding_legend[3],
+                           l = plot_constants$padding_legend[4] + plot_constants$legend_indent,
+                           "bigpts")
```

Comment:
Could plot.margin and legend.margin including legend_indent be applied to theme_cmap? Are they already, to some degree? It would be nice to remove this code from here if possible.

Reply:
@dlcomeaux has pointed out that it is important to allow the user to adjust these items at the finalize_plot stage.